openhcl/underhill_mem/src/mapping.rs

```diff
@@ -17,22 +17,200 @@ use memory_range::MemoryRange;
 use parking_lot::Mutex;
 use sparse_mmap::SparseMapping;
 use std::ptr::NonNull;
+use std::sync::Arc;
 use thiserror::Error;
 use vm_topology::memory::MemoryLayout;
 
+pub struct GuestPartitionMemoryBuilder<'a> {
+    memory_layout: &'a MemoryLayout,
+    valid_memory: Option<Arc<GuestValidMemory>>,
+}
+
+impl<'a> GuestPartitionMemoryBuilder<'a> {
+    /// valid_bitmap_state is valid when allocating a tracking bitmap is needed
```

Comment:
nit: usually we markdown format variables, input parameters with `. so it would look something like `valid_bitmap_state`, since that renders nicer in rustdoc.